### ISSUE TITLE
dynamodb doesn't return initialized attribute on the first update

### ISSUE THREAD

**Issue by chris-erickson** [spulec/moto#1803]
Using DynamoDB, I normally do something like this to update an item (or initialize it to a value if it's unset):

```python
    update_response = table.update_item(
         Key={
             'user': user_id,
         },
         UpdateExpression="ADD num_logins :val",
         ExpressionAttributeValues={
             ':val': decimal.Decimal(1)
         },
         ReturnValues="ALL_NEW",
     )
```
I then get the finalized value with:

```python
number_of_logins = update_response['Attributes']['num_logins']
```

Normally, this is the behavior the first time:

```python
print(number_of_logins)
> 1
```
but with Moto I get a `KeyError` as `update_response['Attributes'][num_logins]` does not exist (`update_response['Attributes']` is `{}`).

So I needed to use this instead:
```python
update_response['Attributes'].get('num_logins', 1)
```

Which is kinda hacky to write code to accommodate tests.

Using (due to the issues with urllib3):
```
boto3==1.8.5
moto==1.3.5
```
and `@mock_dynamodb2`

Happy to contribute a PR but it's not exactly clear to me what I've stumbled on.

**Comment by tvdijk** [comment]
I don't see how this is a feature request. It's a bug, because the behavior of `update_item()` doesn't correctly mimic DynamoDB's behavior.

**Comment by spulec** [comment]
Closed with #1938
